### PR TITLE
Experiment with thinc as a model abstraction

### PR DESCRIPTION
**Pull request by justindujardin**
The reliance on Tensorflow has been alright, but it would be nice to work with a more abstract interface so you could use PyTorch or something else for agent models.

This PR looks at using that latest version of thinc as a model abstraction for Mathy's agents.

Changes:
 - [x] Replace `PolicyValueModel` with `ThincPolicyValueModel` and update usages so agents/train with wrapper
 - [ ] Calculate custom loss and use thinc model update functions instead of unwrapped TF model.
 - [x] Make thinc TensorflowWrapper capable of loading keras models that are subclasses?

Running the code:
The mathy site has an [overview of the build tooling](https://mathy.ai/contributing/), or you can get started quickly by running this from the root:
```bash
sh tools/setup.sh
sh tools/test.sh
```

You can train an agent using the wrapped model:
```bash
mathy train a3c poly training/my_model
```

You can run the thinc_basic.py example:
```
./libraries/mathy_python/.env/bin/python ./libraries/website/docs/snippets/ml/thinc_basic.py 
```

### CODE REVIEW

**Comment by justindujardin** [comment]
Closing this for now because it's out of date
